sklearn/metrics/_ranking.py

```diff
@@ -1733,6 +1733,16 @@ def ndcg_score(y_true, y_score, *, k=None, sample_weight=None, ignore_ties=False
     return np.average(gain, weights=sample_weight)
 
 
+@validate_params(
+    {
+        "y_true": ["array-like"],
+        "y_score": ["array-like"],
+        "k": [Integral],
```

Comment:
why not limit usage to >1
i.e.         "k": [Interval(Integral, 1, None, closed="left"), None],

Reply:
Indeed, `k` cannot be negative. We should use an interval here. However, we cannot use `None` because we expect only integral values.

Reply:
None is fine since ``closed="left"``

Reply:
Ups, I misread. I read that it was an option `None` (not inside the interval). My bad.

Reply:
Ah no you're right. I did misread and didn't see the additional None outside the interval :)

Reply:
So I misread your misreading :)

Reply:
Thank you for all the suggestions and help!